taxonomy/utils.py

```diff
@@ -11,11 +11,11 @@
 LOGGER = logging.getLogger(__name__)
 
 
-def update_skills_data(course_key, confidence, skill_data):
+def update_skills_data(course_key, skill_external_id, confidence, skill_data):
     """
     Persist the skills data in the database.
     """
-    skill, __ = Skill.objects.update_or_create(**skill_data)
+    skill, __ = Skill.objects.update_or_create(external_id=skill_external_id, defaults=skill_data)
```

Comment:
@muhammad-ammar @saleem-latif made changes here as well. please review this part with extra care.